Orange/widgets/data/owfile.py

```diff
@@ -142,6 +146,142 @@ def __repr__(self):
     __str__ = __repr__
 
 
+class VarTableModel(QtCore.QAbstractTableModel):
+    places = "feature", "class", "meta", "skip"
+    typenames = "nominal", "numeric", "string"
+    vartypes = DiscreteVariable, ContinuousVariable, StringVariable
+    name2type = dict(zip(typenames, vartypes))
+    type2name = dict(zip(vartypes, typenames))
+
+    def __init__(self, widget):
+        super().__init__()
+        self.widget = widget
+        self.variables = []
+
+    def set_domain(self, domain):
+        def may_be_numeric(var):
```

Comment:
This function assumes that string variables cannot be numeric: the recognition process is such that if all values can be interpreted as numbers the attribute is numeric. Is this OK or should we scan the actual attribute values in the table?